pyproject.toml

```diff
@@ -3,3 +3,10 @@ profile="jinja"
 linter_output_format = "{filename}:{line}: {code} {message} {match}"
 # Ignore linting rules that are related to web pages
 ignore = "H016,H021,H030,H031"
+
+[tool.ruff]
+extend-exclude = ["data", "models", "db", "cache"]
+output-format = "grouped"
+
+[tool.ruff.lint]
+select = ["E4", "E7", "E9", "F", "I"]
```

Comment:
Why listing individual `E` and not enabling all `E`?

Reply:
I listed them as suggested by @suhaibmujahid in his last comment [here](https://github.com/mozilla/bugbot/pull/2335#discussion_r1479190863)

Reply:
Oh OK. @suhaibmujahid wdyt of enabling all `E`? I would prefer to reduce decisions for linting and just try to rely on safe defaults that most of the community is using too.

Reply:
@marco-c, I enabled all E. 
Code can't pass bugbot tests cause of rules 501 and C901. Should I add them to ignore list?

Reply:
> Why listing individual E and not enabling all E?

This is the default by Ruff (i.e., `["E4", "E7", "E9", "F"]`)

> @suhaibmujahid wdyt of enabling all E? 

Most of the `E` rules are stylistic, which will be handled by the formatter.

> I would prefer to reduce decisions for linting and just try to rely on safe defaults that most of the community is using too.

I agree; I prefer going with the defaults and enabling more case-by-case when it is needed.

> Code can't pass bugbot tests cause of rules 501 and C901. Should I add them to ignore list?

Since `C901` is not passing, it means that it was not enabled. We won't be able to enable it here since fixing that will be out of the scope of this PR.

For `E501`, it was ignored in our `flake8` configs; let us not enable it here.


I would say, let we go with `["E4", "E7", "E9", "F", "I" "T10", "CPY"]`; the following are the justifications: 
- `E4`, `E7`, `E9`, `F`: the defaults for `ruff`
- `I`: replacement for `isort`
- `T10`: replacement for `flake8-debugger`
- `CPY`: replacement for `flake8-copyright`